common/core/desktop/src/kmx/kmx_processevent.cpp

```diff
@@ -33,9 +33,10 @@ KMX_ProcessEvent::~KMX_ProcessEvent() {
 }
 
 char VKeyToChar(KMX_UINT modifiers, KMX_UINT vk) {
-  // We only map SHIFT and UNSHIFTED, and CAPS LOCK
+  // We only map keys that are unmodified, shifted, or caps locked
 
-  if ((modifiers & ~(K_SHIFTFLAG | CAPITALFLAG)) != 0) {
+  // Test for modifier flags excluding Shift, Caps, Num and Scroll Lock
+  if ((modifiers & ~(K_SHIFTFLAG | CAPITALFLAG | NUM_SCROLL_LOCK_MASK)) != 0) {
```

Comment:
I think it might be better to mask out those bits from `modifiers` on input to the function, and document, and simplify the test accordingly:

```
  // We are not interested in Num Lock and Scroll Lock state
  modifiers = modifiers & ~NUM_SCROLL_LOCK_MASK;
```

Reply:
This change should not be necessary if we go with the other fix 😸